client/blocks/comments/post-comment.jsx

```diff
@@ -318,12 +318,26 @@ class PostComment extends PureComponent {
 		} else if ( commentAuthor.site_ID ) {
 			commentAuthorUrl = getStreamUrl( null, commentAuthor.site_ID );
 		} else {
-			commentAuthorUrl = commentAuthor?.URL;
+			const urlToCheck = commentAuthor?.URL;
+			if ( urlToCheck && this.isValidUrl( urlToCheck ) ) {
+				commentAuthorUrl = urlToCheck;
+			}
 		}
 
 		return { comment, commentAuthor, commentAuthorUrl, commentAuthorName };
 	};
 
+	isValidUrl = ( url ) => {
+		try {
+			// URL constructor throws an error if the URL is invalid.
+			const urlObject = new URL( url );
+			// Only allow http or https URLs.
+			return urlObject.protocol === 'http:' || urlObject.protocol === 'https:';
+		} catch ( e ) {
+			return false;
+		}
+	};
+
```

Comment:
It looks like we have a handful of ad-hoc definitions for isValidURL across the app. Would it make sense to use any of those? Some seem to be coming from a 'valid-url' module 🤔

Reply:
Good point! There is an [isURL](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-url/#isurl) function in the wordpress/url package that does nearly the same thing. How about we use that?